hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java

```diff
@@ -252,6 +252,11 @@ private static void setEnvironment(String key, String value) throws Exception {
                 map.put(key, value);
             }
         }
+
+        Field field = Class.forName("java.lang.ProcessEnvironment").getDeclaredField("theCaseInsensitiveEnvironment");
```

Comment:
can you please explain reasoning behind this? 
this whole method is a mystery to me and this adds additional sorcery into the mix:)